netgrasp/netgrasp.py

```diff
@@ -10,17 +10,19 @@
 import logging
 import logging.handlers
 import pwd
-import sys
 import os
 import datetime
 import time
-import traceback
 
 netgrasp_instance = None
 
 BROADCAST = 'ff:ff:ff:ff:ff:ff'
 
-ALERT_TYPES = ['requested_ip', 'first_requested_ip', 'first_requested_ip_recently', 'seen_device', 'first_seen_device', 'first_seen_device_recently', 'seen_mac', 'first_seen_mac', 'seen_ip', 'first_seen_ip', 'seen_host', 'first_seen_host', 'seen_vendor', 'first_seen_vendor', 'device_stale', 'request_stale', 'changed_ip', 'duplicate_ip', 'duplicate_mac', 'network_scan', 'ip_not_on_network', 'src_mac_broadcast', 'requested_self']
+ALERT_TYPES = ['requested_ip', 'first_requested_ip', 'first_requested_ip_recently', 'seen_device', 'first_seen_device',
+               'first_seen_device_recently', 'seen_mac', 'first_seen_mac', 'seen_ip', 'first_seen_ip', 'seen_host',
+               'first_seen_host', 'seen_vendor', 'first_seen_vendor', 'device_stale', 'request_stale', 'changed_ip',
+               'duplicate_ip', 'duplicate_mac', 'network_scan', 'ip_not_on_network', 'src_mac_broadcast',
+               'requested_self']
```

Comment:
Thanks for the kind and thoughtful feedback. All good points and taken in good faith. :) No worries. I'm happy with whatever you're comfortable with.

I've been thinking about the same thing you mentioned above since I commented and had the same thought... am I pushing onto you things PyCharm flags as warnings? I don't want to do that unless they're things you agree with and conform to whatever style you decide.

So PyCharm "flags/highlights" warnings in the editor (pep8 violations, long lines, unused imports/variables, variable names that clash with python built-in names (e.g. "range"), variable names that shadow other names (e.g. function or imported names), etc.). I can individually mute or turn each of these off for the project so they're no longer shown. I'll do that for the things you're not concerned with.

However, I will say they've helped me find bugs and kept things consistent and clean and in general what I'd say is more maintainable code; for my definitions of "consistent", "clean" and "maintainable". :) Your definitions may be different and I respect that. I'll keep an eye out for your definition of those terms and what you're willing to tolerate and adjust accordingly.

What would you think of a "CONTRIBUTING.md" file? Maybe we could create that doc in the project's root dir and work on editing a section named "Project Style Guide"?

BTW, in regards to 2) above, I agree. I'm OCD enough to volunteer to submit a single patch that wraps all long lines in all source files to the style you decide, whenever/whatever that is.

In regards to 3) above, my humble opinion is the gain in readability of seeing all the code on the page at the same time without scrolling allows you to see all the logic/variables and think through/reason about the code and outweighs the added risk.

I don't know how to attach a screenshot in these comments, but I'm curious to see how vim wraps for you. I was thinking that for you, from what you're describing, vim would wrap to the next line and wouldn't preserve the leading/block-level whitespace? If that's the case, I'd think that would be more difficult to read. I must be thinking about this wrong and I can't visualize what you're seeing. For long lines, for me in PyCharm, it's all one, single line that I have to scroll horizontally to read.